pkg/controller/import-controller.go

```diff
@@ -439,9 +439,17 @@ func (r *ImportReconciler) updatePvcFromPod(pvc *corev1.PersistentVolumeClaim, p
 }
 
 func (r *ImportReconciler) cleanup(pvc *corev1.PersistentVolumeClaim, pod *corev1.Pod, log logr.Logger) error {
-	if err := r.client.Delete(context.TODO(), pod); cc.IgnoreNotFound(err) != nil {
+	deleteOptions := &client.DeleteOptions{}
+	// Setting grace period to 0 if pod already has deletion timestamp.
+	// This allows avoiding race conditions during pod termination.
+	if pod.DeletionTimestamp != nil {
+		gracePeriod := int64(0)
+		deleteOptions.GracePeriodSeconds = &gracePeriod
```

Comment:
Can probably do `deleteOptions.GracePeriodSeconds = pointer.Int64Ptr(0)` instead.

Reply:
Actually following https://github.com/kubevirt/containerized-data-importer/pull/3009/commits/aa123c07faf5224ffb81cbfe1805c4eb707b6b0b it's `ptr.To[int64](0)`

Reply:
Either way works, looks like we have both in the code base. I would suggest we standardize on one.

Reply:
Where? we shouldn't have `k8s.io/utils/pointer` anymore

Reply:
Oh I had not pulled the latest, after rebase I see no trace of `pointer.`